helix-core/src/main/java/org/apache/helix/controller/stages/PerReplicaThrottleStage.java

```diff
@@ -270,4 +539,24 @@ public void setPriority(String priority) {
       }
     }
   }
+
+  private boolean isDownwardTransition(IdealState idealState, ResourceControllerDataProvider cache,
+      Message message) {
+    boolean isDownward = false;
+
+    String stateModelDefName = idealState.getStateModelDefRef();
+    StateModelDefinition stateModelDef = cache.getStateModelDef(stateModelDefName);
+    Map<String, Integer> statePriorityMap = stateModelDef.getStatePriorityMap();
+    String fromState = message.getFromState();
+    String toState = message.getToState();
+    if (statePriorityMap.containsKey(fromState) && statePriorityMap.containsKey(toState)) {
```

Comment:
change to following comment:

// Only when both fromState and toState can be found in the statePriorityMap, comparision of
    // state priority is used to determine if the transition is downward. Otherwise, we can't
    // really determine the order and by default consider the transition not downard.